erts/emulator/beam/jit/arm/32/instr_fun.cpp

```diff
@@ -85,11 +85,11 @@ void BeamGlobalAssembler::emit_apply_fun_shared() {
     Label finished = a.newLabel();
 
     /* Put the arity and fun into the right registers for `call_fun`, and stash
-     * the argument list in ARG5 for the error path. We'll bump the arity as
+     * the argument list in ARG4 for the error path. We'll bump the arity as
      * we go through the argument list. */
-    mov_imm(ARG3, 0);
-    //a.mov(ARG4, getXRef(0));
-    //a.mov(ARG5, getXRef(1));
```

Comment:
Didn't we discuss to keep ARG3 and ARG4 and replace ARG5 with ARG2. It would make the diff to the  ARM 64 code that we base our implementation on less complex.

Reply:
I do not think we should mirror the code and keep difss to minimum. BTW there is no real "diff to arm64 code" since our solution will look very different. I think we should understand ARM64  and replicate it semantically 1to1 in arm32 but we should not try to mirror it. 
They used 3 4 and 5 because of natural incremental order, we do not have 5 so we use 2, 3, 4, that looked natural to me...

Reply:
There even is a diff arm64 to x86. I look at it all the time and like that I can do that to pick from both. Our solution will not look that different from the two existing ones.

Reply:
Ok let's go for it